### PR TITLE
Support websocket subprotocols

### PR DESCRIPTION
**Pull request by JacobChang**
this should fix issue #355 and also issue #593

### CODE REVIEW

**Comment by haf** [comment]
@JacobChang Thank you for your pull request! We'll review your code shortly.

**Comment by haf** [comment]
@ademar I don't see any problems merging this, but I also haven't read the WebSocket spec. I think you should review this.

File: src/Suave/WebSocket.fs [hunk 1]
```diff
@@ -275,43 +279,77 @@ module WebSocket =
...
+        | _ -> Async.result None
+
+      let! a = handShakeAux subprotocol webSocketKey continuation ctx
+      match a with
+      | Choice1Of2 _ ->
+        do ()
+      | Choice2Of2 err ->
+        ctx.runtime.logger.info (
+          eventX "WebSocket disconnected"
+          >> setSingleName "Suave.Websocket.handShake"
```
Comment (ademar) [inline src/Suave/WebSocket.fs]: please change to
```
>> setSingleName "Suave.Websocket.handShakeWithSubprotocol"
```

**Comment by ademar** [comment]
Looks good to me :+1:

File: src/Suave/WebSocket.fs (hunk 1 above)
Comment (JacobChang) [reply src/Suave/WebSocket.fs]: @ademar comments are addressed

**Comment by haf** [comment]
Pending for v2.3.0-beta.2

**Pull request merged**